src/mcode-pilot/components/OptionsCheckboxList/OptionsCheckboxList.jsx

```diff
@@ -44,9 +60,40 @@ export default class OptionsCheckboxList extends Component {
                                             value={displayText}
                                             className="checkbox" />
                                     }
-                                    label={<span className="selection-options__title">{optionText}</span>}
+                                    label={
+                                        <span className="selection-options__title">
+                                            {!hasRange && optionText}
+                                            {hasRange && <span className="display-text">{displayText}: </span>}
+                                        </span>
+                                    }
                                 />
                             </FormLabel>
+
+                            {hasRange &&
+                                <div>
+                                    <span className="range" onClick={() => this.handleOpenRangeSelector(key)}>
+                                        {minValue}-{maxValue}
+                                    </span>
+                                    {openRangeSelector[key] &&
+                                        <ClickAwayListener onClickAway={() => this.handleCloseRangeSelector(key)}>
+                                            <OptionsRangeSelector
+                                                className={displayText}
+                                                name={displayText}
+                                                unit={unit}
+                                                value={value}
+                                                min={minValue}
+                                                max={maxValue}
+                                                defaultMin={defaultMinValue}
+                                                defaultMax={defaultMaxValue}
+                                                patientAge={patientAge}
+                                                category={category}
+                                                keyy={key}
```

Comment:
Is there a reason this is keyy and not key?

Reply:
https://reactjs.org/warnings/special-props.html ... I can rename it to something else if you have strong objections